### PR TITLE
Don't error out for variadic functions in Do/DoReturn

### PR DESCRIPTION
**Pull request by joshblum**
Addresses https://github.com/golang/mock/issues/600

### CODE REVIEW

**Comment by joshblum** [comment]
@googlebot I signed it!

**Comment by codyoss** [comment]
Discussing options on issue. Holding review for now.

**Comment by joshblum** [comment]
@codyoss updated the error messaging here, happy to change the copy to what you think is most clear

File: gomock/call.go [hunk 1]
```diff
@@ -144,20 +149,25 @@ func (c *Call) DoAndReturn(f interface{}) *Call {
...
 
 	c.addAction(func(args []interface{}) []interface{} {
 		c.t.Helper()
-		if c.methodType.NumIn() != v.Type().NumIn() {
-			c.t.Fatalf("wrong number of arguments in Do func for %T.%v: got %d, want %d [%s]",
-				c.receiver, c.method, v.Type().NumIn(), c.methodType.NumIn(), c.origin)
+		ft := v.Type()
+		if c.methodType.NumIn() != ft.NumIn() {
+			if ft.IsVariadic() {
+				c.t.Fatalf("wrong number of arguments in DoAndReturn func for %T.%v The function signature must match the mocked method, a variadic function cannot be used.",
```
Comment (codyoss) [inline gomock/call.go:162]: Please update error message to be about Do instead of DoAndReturn

**Review by codyoss — CHANGES_REQUESTED** [review]
Just one small thing!

File: gomock/call.go (hunk 1 above)
Comment (joshblum) [reply gomock/call.go]: ah thanks for catching!

**Review by codyoss — APPROVED** [review]
LGTM

**Pull request merged**